src/python/foglamp/backup_restore/restore.py

```diff
@@ -0,0 +1,693 @@
+    _STATUS_RUNNING = 2
+
+    _FOGLAMP_CMD = "python3 -m foglamp {0}"
+    """ Command for managing FogLAMP, stop/start/status """
+
+    _config_from_manager = {}
+    _config = {}
+
+    _logger = ""
+    _event_loop = ""
+    _job = ""
+
+    def __init__(self):
+        self._config_from_manager = {}
+        self._config = {}
+
+        self._job = lib.Job()
+        self._event_loop = asyncio.get_event_loop()
+
+        try:
+            self._logger = logger.setup(self._MODULE_NAME)
+
+            # Sets the logger for the library
+            lib._logger = self._logger
+
+        except Exception as _ex:
+            _message = self._MESSAGES_LIST["e000001"].format(str(_ex))
+            _current_time = time.strftime("%Y-%m-%d %H:%M:%S")
+
+            print("[FOGLAMP] {0} - ERROR - {1}".format(_current_time, _message), file=sys.stderr)
+            sys.exit(1)
+
+    def _foglamp_stop(self):
+        """ Stops FogLAMP before for the execution of the backup, doing a cold backup
+
+        Args:
+        Returns:
+        Raises:
+            FogLAMPStopError
+        Todo:
+        """
+
+        self._logger.debug("{func}".format(func=sys._getframe().f_code.co_name))
+
+        cmd = self._FOGLAMP_CMD.format("stop")
+
+        # Restore the backup
+        status, output = lib.exec_wait_retry(cmd, True,
+                                             max_retry=self._config['max_retry'],
+                                             timeout=self._config['timeout'])
+
+        self._logger.debug("{func} - status |{status}| - cmd |{cmd}| - output |{output}|   ".format(
+                    func=sys._getframe().f_code.co_name,
+                    status=status,
+                    cmd=cmd,
+                    output=output))
+
+        if status == 0:
+            if self._foglamp_status() != self._STATUS_STOPPED:
+                raise FogLAMPStopError(output)
+        else:
+            raise FogLAMPStopError(output)
+
+    def _foglamp_start(self):
+        """ Starts FogLAMP after the execution of the restore
+
+        Args:
+        Returns:
+        Raises:
+            FogLAMPStartError
+        Todo:
+        """
+
+        cmd = self._FOGLAMP_CMD.format("start")
+
+        status, output = lib.exec_wait_retry(cmd, True,
+                                             max_retry=self._config['max_retry'],
+                                             timeout=self._config['timeout'])
+
+        self._logger.debug("FogLAMP {0} - output |{1}| -  status |{2}|  ".format(
+                                                                            sys._getframe().f_code.co_name,
+                                                                            output,
+                                                                            status))
+
+        if status == 0:
+            if self._foglamp_status() != self._STATUS_RUNNING:
+                raise FogLAMPStartError
+
+        else:
+            raise FogLAMPStartError
+
```

Comment:
Why start/stop/status from python3 commands from a python script?
Import server_daemon.py and use start/stop/stats directly